index.js

```diff
@@ -8,25 +8,28 @@ const rm = require('rimraf');
 
 module.exports = {
   install: () => {
-       nodeNightlyVersion().then(latest => {
-        const conf = new Configstore(pkg.name);
+  		let osArchString;
+      nodeNightlyVersion().then(latest => {
+      	const conf = new Configstore(pkg.name);
         conf.set('version', latest);
         const os = process.platform;
         const arch = process.arch;
-        const nightlyUrl = process.env.NODEJS_ORG_NIGHTLY_MIRROR || 'https://nodejs.org/download/nightly';
-        const url = `${nightlyUrl}/${latest}/node-${latest}-${os}-${arch}.tar.gz`;
-        download(url, __dirname, {extract:true})
-        .then( _ => {
-          mv(`${__dirname}/node-${latest}-${os}-${arch}`, `${__dirname}/node-nightly`);
-          console.log('node-nightly is available on your CLI! ');
-          process.exit(0);
-        }).catch(err => console.error(err));
-      }).catch(err => console.error(err))
+        const type = 'nightly';
+        osArchString = `${latest}-${os}-${arch}`;
+        const url = `https://nodejs.org/download/${type}/${latest}/node-${osArchString}.tar`;
+        return download(url, __dirname, {extract:true});
+      })
+      .then( _ => {
+      	mv(`${__dirname}/node-${osArchString}`, `${__dirname}/node-nightly`);
+        console.log('node-nightly is available on your CLI! ');
+        process.exit(0);
+      })
+      .catch(console.error);
   },
-  update: function() {
```

Comment:
Uggh!
I think i got it ..
Is it because arrow function's `this` is picked up from surroundings (lexical) not the object from which it is being called upon.
I think it's a gotcha of `Arrow function`. Correct me if wrong :surfer: